### ISSUE TITLE
[tsql] Omitted comma between column list and constraint leads to PRS ("Found unparsable section")

### ISSUE THREAD

**Issue by medianick** [sqlfluff/sqlfluff#6340]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

With a file like the following, I get a PRS ("Found unparsable section") error:

```sql
CREATE TABLE [dbo].[Foo](
    [ID] [int] IDENTITY(1,1) NOT NULL
    CONSTRAINT [PK_Foo_ID] PRIMARY KEY CLUSTERED
    (
        [ID] ASC
    )
)
```

I believe what is happening is that Sqlfluff is requiring that there be a comma separating the column from the PK constraint, but SQL Server does not require this.

### Expected Behaviour

I expected this to be parsable.

### Observed Behaviour

This raised an error:

```
== [Test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'CREATE
                       | TABLE [dbo].[Foo](\n\t[ID] [int] ID...'
WARNING: Parsing errors found and dialect is set to 'tsql'. Have you configured your dialect correctly?
```

### How to reproduce

See included script.

### Dialect

`tsql`

### Version

3.2.3

### Configuration

```ini
[sqlfluff]
dialect = tsql
exclude_rules = CP01,CP02,CP03,CP05,LT01,LT02,LT05,LT12,LT13,RF06

[sqlfluff:rules:references.quoting]
prefer_quoted_identifiers = True

[sqlfluff:rules:references.special_chars]
allow_space_in_identifier = True
additional_allowed_characters = .
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
